package/generate_spec.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+"""
+This RPM spec generator creates a distribution-specific .spec file.
+It extracts versions from Cargo.toml/rust-toolchain.toml and scans git submodules for bundled library declarations.
+
+The script takes a target distribution, loads the corresponding template,
+and outputs a complete RPM spec file ready for rpmbuild.
+"""
+
+import argparse
+import subprocess
+from pathlib import Path
+from datetime import datetime
+from jinja2 import Environment, FileSystemLoader
+import tomllib
+
+script_dir = Path(__file__).parent
+project_root = script_dir.parent
+templates_dir = script_dir / "templates"
+
+
+def get_version() -> str:
+    cargo_path = project_root / "mountpoint-s3" / "Cargo.toml"
+    with open(cargo_path, "rb") as f:
+        data = tomllib.load(f)
+        return data["package"]["version"]
+
+
+def get_rust_version() -> str:
+    rust_path = project_root / "rust-toolchain.toml"
+    with open(rust_path, "rb") as f:
+        data = tomllib.load(f)
+        return data["toolchain"]["channel"]
+
+
+def get_submodule_versions() -> dict[str, str]:
+    result = subprocess.run(
+        'git submodule foreach -q \'echo $name `git describe --tags`\'', capture_output=True, text=True, shell=True
+    )
+    versions = {}
+    for line in result.stdout.strip().split('\n'):
+        match line.split(' ', 1):
+            case [name, version]:
+                versions[name] = version.removeprefix('v')
+    return versions
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Generate RPM spec files for different distributions")
+    parser.add_argument("build_target", help="Target distribution (e.g., amzn2023)")
+    parser.add_argument("--template", help="Custom template file (default: {build_target}.spec.template)")
```

Comment:
I'd apply some YAGNI here and start with something simpler for the specific case we are targeting now.